### PR TITLE
Persistent loglevel

### PR DESCRIPTION
**Pull request by EdenGri**
This PR needs to be merged only after the following PR: https://github.com/dprital/sonic-buildimage/pull/12
This PR needs to be merged with this PR: https://github.com/EdenGri/sonic-utilities/pull/7

What I did
Add the ability to the user to save the loglevel and make it persistent to reboot.

How I did it
Move the logger tables from the LOGLEVEL DB to the CONFIG DB

How to verify it
1. change the orchagent loglevel (for example) -> swssloglevel -c orchagent -l DEBUG
2. save the loglevel -> run config save
3. reboot
4. verify that the orchagent log level is still DEBUG ->run run redis-cli -n 4 hgetall "LOGGER|orchagent"

### CODE REVIEW

File: common/logger.cpp [hunk 1]
```diff
@@ -193,27 +199,19 @@ Logger::Priority Logger::getMinPrio()
...
+
 void Logger::settingThread()
 {
     Select select;
-    DBConnector db("LOGLEVEL_DB", 0);
-    std::map<std::string, std::shared_ptr<ConsumerStateTable>> selectables;
+    DBConnector db("CONFIG_DB", 0);
+
+    std::map<std::string, std::shared_ptr<SubscriberStateTable>> selectables;
+    auto table = std::make_shared<SubscriberStateTable>(&db, "LOGGER");
```
Comment (DavidZagury) [inline common/logger.cpp]: Can also be defined

File: tests/Makefile.am [hunk 2]
```diff
@@ -28,6 +28,8 @@ tests_SOURCES = redis_ut.cpp                \
                 warm_restart_ut.cpp         \
                 redis_multi_db_ut.cpp       \
                 logger_ut.cpp               \
+                ../common/loglevel.cpp		\
```
Comment (DavidZagury) [inline tests/Makefile.am]: Switch to spaces

File: common/loglevel.cpp [hunk 3]
```diff
@@ -190,8 +211,11 @@ int main(int argc, char **argv)
...
         exitWithUsage(EXIT_FAILURE, "Component not present in DB");
     }
-
-    setLoglevel(db, component, loglevel);
+    setLoglevel(logger_tbl, component, loglevel);
 
     return EXIT_SUCCESS;
 }
+
+
```
Comment (DavidZagury) [inline common/loglevel.cpp]: Remove

File: tests/loglevel_ut.cpp [hunk 4]
```diff
@@ -0,0 +1,43 @@
+#include "common/dbconnector.h"
+#include "common/producerstatetable.h"
+#include "common/consumerstatetable.h"
+#include "common/select.h"
+#include "common/schema.h"
+#include "common/loglevel.h"
+#include "logger_ut.h"
+#include "gtest/gtest.h"
+#include <unistd.h>
+using namespace std;
```
Comment (DavidZagury) [inline tests/loglevel_ut.cpp]: Try to remove the using namespace

**Review by DavidZagury — CHANGES_REQUESTED** [review]

File: common/loglevel.cpp [hunk 5]
```diff
@@ -72,14 +68,35 @@ bool filterSaiKeys(const std::string& key)
...
+    return keys;
+}
+
+void setAllLoglevel(swss::Table& logger_tbl, std::vector<std::string> components,std::string loglevel)
+{
+    for (const auto& component : components)
+    {
+        setLoglevel(logger_tbl, component, loglevel);
+    }
+    SWSS_LOG_DEBUG("All components are with default loglevel");
```
Comment (dprital) [inline common/loglevel.cpp]: Please fix this comment. It might be set to other value

**Pull request closed without merging**